2024_scraper_national_test.py

```diff
@@ -0,0 +1,223 @@
+import requests
+import json
+import re
+from bs4 import BeautifulSoup #data scraper package
+import numpy
+import pandas as pd
+from itertools import zip_longest
+#url loading and get
+url = 'https://www.realclearpolling.com/polls/president/general/2024/trump-vs-biden#polls'
+page = requests.get(url)
+
+#check and return code
+if page.status_code == 200:
+    url_content = page.text
+    print("Success")
+else:
+    print("Failed to retrieve. Status Code was", page.status_code)
+    exit()
+
+soup = BeautifulSoup(page.content, "html.parser") #EVERYTHING
+
+script_tags = soup.find_all('script')
+
+counter = 0 #this checks the line with the JS that has ALL THE DATA in it
+for script in script_tags:
+    if script.string and 'finalData' in script.string:
+        counter += 1
+        #print(script.string)
+        str = script.string
+        break
+#shaving the values into readable json for json.loads
+str2 = str.split('self.__next_f.push(')
+str3 = str2[1][:-1]
+json = json.loads(str3)
+#print(json[1]) #so now its in a big LIST
+json_str = json[1] #chooses the 2nd item in the list indexed as 1
+print(json_str)
```

Comment:
this is not working as expected. the json.loads function is taking the data and converting it into a list of two huge strings.

you can do regex like the way you're doing it, or you can make sure that these strings are ALSO converted to jsons and in the correct json format. something is off about them if each of the strings inside are not being converted into either a python dictionary or a python list.

Reply:
print test produces this for the maryland data with all the escape slashes gone.
[JSON.LOADS.txt](https://github.com/btparrish/election2024-highd-modeling/files/15477075/JSON.LOADS.txt)

JSON.loads produces this which is sufficient enough for the regex to comb through.

im going to upload a former file without all the fucky dictionary WIP business but it does infact run with the maryland state National Url

Reply:
https://github.com/Tooprotimmy/election2024-highd-modeling/blob/patch-1/JSON.LOADS%20ticket.py

Reply:
![image](https://github.com/btparrish/election2024-highd-modeling/assets/156129036/57fd1752-d1fe-4c6a-b2f9-ada52d34c2c8)

Reply:
https://github.com/Tooprotimmy/election2024-highd-modeling/blob/patch-1/JSON.LOADS%20ticket.py

updated test scraper. had to add new regex patterns for rvalue/dvalues (percentage in polls) as different websites can have different respective patterns.